neo/rawio/axonarawio.py

```diff
@@ -83,55 +100,122 @@ def _parse_header(self):
-        self.global_header_size = 0
-
-        # Generally useful information
-        self.sr = int(params['rawRate'])
-        self.num_channels = len(self.get_active_tetrode()) * 4
-
-        # How many 432byte packets does this data contain (<=> num. samples/3)?
-        self.num_total_packets = int(os.path.getsize(self.bin_file)
-                                     / self.bytes_packet)
-        self.num_total_samples = self.num_total_packets * 3
-
-        # Create np.memmap to .bin file
-        self._raw_signals = np.memmap(
-            self.bin_file, dtype=self.data_dtype, mode='r',
-            offset=self.global_header_size
-        )
-
-        # Header dict
+        global_params = self.get_header_parameters(self.set_file, params)
+
+        unit_dtype = np.dtype([('spiketime', '>i4'),
+                               ('samples', 'int8', (50,))])
+
+        # Utility collection of file parameters (general info and header data)
+        params = {'bin': {'bytes_packet': 432,
+                          'bytes_data': 384,
+                          'bytes_head': 32,
+                          'bytes_tail': 16,
+                          'data_type': np.int16,
+                          'header_size': 0,
+                          'header_encoding': 'cp1252',
+                          'sampling_rate': int(global_params['rawRate']),
+                          'num_channels': len(self.get_active_tetrode()) * 4},
+                  'set': {'header_encoding': 'cp1252'},
+                  'unit': {'data_type': unit_dtype,
+                           'wf_left_sweep_us': 200,
+                           'tetrode_ids': []}}
+        self.file_parameters = params
+
+        # SCAN BIN FILE
+        signal_streams = []
+        signal_channels = []
+        if self.bin_file:
+            # add derived parameters from bin file
+            num_tot_packets = int(
+                self.bin_file.stat().st_size / self.file_parameters['bin'][
+                    'bytes_packet'])
+            self.file_parameters['bin']['num_total_packets'] = num_tot_packets
+            self.file_parameters['bin'][
+                'num_total_samples'] = num_tot_packets * 3
+
+            # Create np.memmap to .bin file
+            self._raw_signals = np.memmap(
+                self.bin_file, dtype=self.file_parameters['bin']['data_type'],
+                mode='r', offset=self.file_parameters['bin']['header_size']
+            )
+
+            signal_streams = self._get_signal_streams_header()
+            signal_channels = self._get_signal_chan_header()
+
+        # SCAN TETRODE FILES
+        # In this IO one tetrode corresponds to one unit
+        self._raw_spikes = []
+        spike_channels = []
+        if self.tetrode_files:
+
+            for i, tetrode_file in enumerate(self.tetrode_files):
+                # collecting tetrode specific parameters and dtype conversions
+                tdict = self.get_header_parameters(tetrode_file, None)
```

Comment:
Yes, that is probably easier to maintain. I added a utility method that is now used whenever a original value needs conversion instead of the precalculation.